crates/ruff_linter/src/message/json.rs

```diff
@@ -74,19 +75,19 @@ pub(crate) fn message_to_json_value(message: &Message, context: &EmitterContext)
         );
         start_location = notebook_index.translate_location(&start_location);
         end_location = notebook_index.translate_location(&end_location);
-        noqa_location = notebook_index.translate_location(&noqa_location);
+        noqa_location = noqa_location.map(|location| notebook_index.translate_location(&location));
     }
 
     json!({
-        "code": message.kind.rule().noqa_code().to_string(),
-        "url": message.kind.rule().url(),
-        "message": message.kind.body,
+        "code": message.rule().map(|rule| rule.noqa_code().to_string()),
```

Comment:
Nit: I'm inclined to add a `code` and `url` method to `Message` that return `Option<String>` and `Option<Url>` to avoid the repetition in the different emitters.